### PR TITLE
feat: show qr codes

### PR DESCRIPTION
**Pull request by richard-ramos**

### CODE REVIEW

File: src/app/profile/qrcode/qrcode.nim [hunk 1]
```diff
@@ -0,0 +1,21 @@
+import strformat
+import strutils
+import qrcodegen
```
Comment (jrainville) [inline src/app/profile/qrcode/qrcode.nim:3]: Could put them on a single line

File: src/app/profile/qrcode/qrcode.nim [hunk 2]
```diff
@@ -0,0 +1,21 @@
...
+    raise newException(Exception, "Error generating QR Code")
+  else:
+    var parts: seq[string] = @[]
+    let size = qrcodegen_getSize(qr0[0].addr);
+    for y in countup(0, size):
+      for x in countup(0, size):
+        if qrcodegen_getModule(qr0[0].addr, x.cint, y.cint):
+          parts.add(&"M{x + border},{y + border}h1v1h-1z")
+    let partsStr = parts.join(" ")
+    result = &"<?xml version=\"1.0\" encoding=\"UTF-8\"?><!DOCTYPE svg PUBLIC \"-//W3C//DTD SVG 1.1//EN\" \"http://www.w3.org/Graphics/SVG/1.1/DTD/svg11.dtd\"><svg xmlns=\"http://www.w3.org/2000/svg\" version=\"1.1\" viewBox=\"0 0 {size + border * 2} {size + border * 2}\" stroke=\"none\"><rect width=\"100%\" height=\"100%\" fill=\"#FFFFFF\"/><path d=\"{partsStr}\" fill=\"#000000\"/></svg>"
```
Comment (jrainville) [inline src/app/profile/qrcode/qrcode.nim:19]: God this is complex for no reason lol. As long as it works 🤷

File: ui/app/AppLayouts/Chat/components/ProfilePopup.qml [hunk 3]
```diff
@@ -63,114 +65,138 @@ ModalPopup {
...
+        fillMode: QtQuick.Image.PreserveAspectFit
+        source: profileModel.qrCode(fromAuthor)
+        anchors.horizontalCenter: parent.horizontalCenter
+        anchors.verticalCenter: parent.verticalCenter
+        height: 212
+        width: 212
+        mipmap: true
+        smooth: false
+      }
     }
```
Comment (jrainville) [inline ui/app/AppLayouts/Chat/components/ProfilePopup.qml:117]: The indentation seems a bit messed up in this file. Our QML files use 4 spaces

Comment (jrainville) [reply ui/app/AppLayouts/Chat/components/ProfilePopup.qml:117]: FYI, QT Creator as a tool to reformat the files in `Tools/QML/reformat`

File: src/app/profile/qrcode/qrcode.nim (hunk 2 above)
Comment (richard-ramos) [reply src/app/profile/qrcode/qrcode.nim:17]: Me while translating that code from TS to Nim:

**Pull request merged**